### PR TITLE
Disable chatroom creation if cell_chknochat

### PR DESCRIPTION
**Pull request by Emistry**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- doesn't allow chatroom creation/switch if the coordinated cell are set with `cell_nochat`.
- chatroom will be auto deleted.

<details>

```
-	script	sample	-1,{

	OnInit:
		.@map$ = "prontera";
		
		.@max_x = getmapinfo(MAPINFO_SIZE_X, .@map$);
		.@max_y = getmapinfo(MAPINFO_SIZE_Y, .@map$);
		setcell(.@map$, 0, 0, .@max_x, .@max_y, cell_nochat, true);
		
		.@range = 10;
		.@center_x = 155;
		.@center_y = 181;
		setcell(.@map$, (.@center_x - .@range), (.@center_y - .@range), (.@center_x + .@range), (.@center_y + .@range), cell_nochat, false);
		end;
}
```
player 1 - create a normal chatroom at normal cells that dont have `cell_nochat`.
player 2 - move to `cell_nochat` area, and join the chatroom
player 1 - leave the chatroom
** at this point, chatroom shall be auto deleted since the new owner are located in the `cell_nochat` area.
</details>

**Issues addressed:** 
https://github.com/HerculesWS/Hercules/issues/1569

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2443)

**Comment by bWolfie** [comment]
Nice one! Much better solution than what I was using where player on nochat cell just can't join the chat.

File: src/map/chat.c [hunk 1]
```diff
@@ -255,6 +255,14 @@ static int chat_leavechat(struct map_session_data *sd, bool kicked)
 	}
 
 	if( leavechar == 0 && cd->owner->type == BL_PC ) {
+
+		// check if new location are CELL_CHKNOCHAT
+		if (map->getcell(cd->usersd[0]->bl.m, NULL, cd->usersd[0]->bl.x, cd->usersd[0]->bl.y, CELL_CHKNOCHAT)) {
+			for (i = (cd->users - 1); i >= 0; i--)
```
Comment (hemagx) [inline src/map/chat.c]: Is there a reason the loop have to be descending instead of ascending?

Comment (Emistry) [reply src/map/chat.c]: doesn't have to shift the array.
just my own preference when delete data from array.

previously i tried with `chat->npckickall(...)` but no idea, it crashed for unknown reason.
btw...  `chat->npckickall(...)` also did go through the loop descending when kicking member, maybe the same reason as stated above.

Comment (hemagx) [reply src/map/chat.c]: Ah, this makes perfect sense then.

**Pull request merged**